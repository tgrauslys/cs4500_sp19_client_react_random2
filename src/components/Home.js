import React from 'react'
import ServiceCategoryPills from "./ServiceCategoryPills";
import ServiceSearchContainer from "../containers/ServiceSearchContainer";
import UserService from "../services/UserService";

const userService = UserService.getInstance()

const Home = (pillServiceCategories) => {
    return (
    <div>
        <h2>Home</h2>
        <div className="row">
            <div className="col-8">
                <h1>
                    Find professionals near you.
                </h1>
                {/*<SearchBar history={history}/> */}
            </div>
            <div className="col-3 text-right">
                <a href="#">Sign up</a>
            </div>
            <div className="col-1">
                <a href="#">Log in</a>
            </div>
        </div>

        <br/>
        <div>
            <ServiceSearchContainer
                service={userService}/>
        </div>
        <br/>
        <br/>
        <br/>
        <div>
            <ServiceCategoryPills serviceCategories={pillServiceCategories.pillServiceCategories}/>
        </div>
        <br/>
        <br/>
        <br/>
        {/*<ServiceTabNavigator
            serviceCategories={serviceCategories}/>*/}
    </div>
)
}
export default Home
