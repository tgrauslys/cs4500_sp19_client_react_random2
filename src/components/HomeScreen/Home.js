import React from 'react'
import ServiceCategoryPills from "./ServiceCategoryPills";
import ServiceSearchContainer from "../../containers/ServiceSearchContainer";
import UserService from "../../services/UserService";
import ServiceNavigator from "../ServiceNavigator/ServiceNavigator";

const userService = UserService.getInstance()
const services = ServicesService.getInstance()

const Home = (pillServiceCategories) => {
    console.log(pillServiceCategories)
    return (
    <div>
        <h2>Home</h2>
        <div className="row">
            <div className="col-8">
                <h1>
                    Find professionals near you.
                </h1>
                {/*<SearchBar history={history}/> */}
            </div>
            <div className="col-3 text-right">
                <a href="#">Sign up</a>
            </div>
            <div className="col-1">
                <a href="#">Log in</a>
            </div>
        </div>

        <br/>
        <div>
            <ServiceCategoryPills serviceCategories={pillServiceCategories.pillServiceCategories}/>
        </div><br/>
        <div>
            <ServiceSearchContainer
                serviceService={services}
                userService={userService}/>
        </div>
        <br/>
        <div>
            <ServiceNavigator
                serviceCategories={pillServiceCategories.pillServiceCategories}
            />
        </div>

    </div>
)
}
export default Home
