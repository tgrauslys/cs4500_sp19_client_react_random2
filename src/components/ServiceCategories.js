import React from 'react'
import {Link} from 'react-router-dom'

const ServiceCategories = ({
                               categories,
                               createCategory,
                               deleteCategory,
                               optionValues,
                               currentPage,
                               itemCount,
                               totalPages,
                               setPage
                           }) => {
    const isFirstPage = currentPage === 0;
    const isLastPage = currentPage >= totalPages - 1;

    let previousButton = <button onClick= {(e) => setPage(e, currentPage - 1)}
                                 style={{margin: '2px'}}
        // &#60; displays "<"
                                 type="button" className="btn btn-secondary">&#60;</button>
    let previousPageButton = <button onClick= {(e) => setPage(e, currentPage - 1)}
                                     style={{margin: '2px'}}
                                     type="button" className="btn btn-primary">{currentPage}</button>
    let nextButton = <button onClick= {(e) => setPage(e, currentPage + 1)}
                             style={{margin: '2px'}}
        // &#62; displays ">"
                             type="button" className="btn btn-secondary">&#62;</button>
    let nextPageButton = <button onClick= {(e) => setPage(e, currentPage + 1)}
                                 style={{margin: '2px'}}
                                 type="button" className="btn btn-primary">{currentPage + 2}</button>

    if (isFirstPage) {
        previousButton = ''
        previousPageButton = ''
    }
    if (isLastPage) {
        nextButton = ''
        nextPageButton = ''
    }


    return (
        <div>
            <h3>Service Categories</h3>
            <a role="button" className="btn btn-success" variant="outline-success"
               onClick={createCategory}
            >Create new</a>
            <table className="table">
                <tbody>
                {
                    categories
                        .map(serviceCategory =>
                                 <tr key={serviceCategory.id}>
                                     <td>
                                         <Link to={`/admin/categories/${serviceCategory.id}`}>
                                             {serviceCategory.serviceCategoryName}
                                         </Link>
                                     </td>
                                     <td>
                                         <a role="button" className="btn btn-danger"
                                            onClick={() => {
                                                deleteCategory(
                                                    serviceCategory.id)
                                            }}>
                                             Delete
                                         </a>
                                     </td>
                                 </tr>
                        )
                }
                <tr>
                    <td>
                        <select value={itemCount} onChange={e => setPage(e)}>
                            {
                                optionValues.map(possibleItemCounts =>
                                                     <option key={possibleItemCounts} value={possibleItemCounts}>{possibleItemCounts}</option>)
                            }
                        </select>
                        {previousButton}
                        {previousPageButton}
                        <button
                            style={{margin: '2px'}}
                            type="button" className="btn btn-primary" disabled>{currentPage + 1}</button>
                        {nextPageButton}
                        {nextButton}
                    </td>
                </tr>
                </tbody>
            </table>
        </div>)
}
export default ServiceCategories