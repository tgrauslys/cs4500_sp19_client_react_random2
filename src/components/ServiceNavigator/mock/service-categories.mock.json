[
  {
    "id": "123", "title": "Home Improvements",
    "services": [
      {"id": "1", "title": "Appliance Installation"},
      {"id": "2", "title": "Cabinetry"},
      {"id": "3", "title": "Door Installation"},
      {"id": "4", "title": "Concrete Removal"},
      {"id": "5", "title": "Dishwasher Installation"},
      {"id": "6", "title": "Finish Carpentry"},
      {"id": "7", "title": "Floor Polishing"},
      {"id": "8", "title": "Furniture Assembly"},
      {"id": "9", "title": "Generator Installation"},
      {"id": "10", "title": "Gardening"},
      {"id": "11", "title": "Home Organizing"},
      {"id": "12", "title": "Electrical and Wiring Repair"}
    ],
    "serviceProviders": [
      {
        "title": "Christina's Custom Cleaning, CCC",
        "rating": 4.4,
        "ratingScores": [
          93,
          24,
          12,
          10,
          4
        ],
        "yearsInBusiness": "7",
        "hires": 17,
        "latestReview": "It's so wonderful to have the burden of cleaning - especially with an injury - removed. Christina is excellent with communication and I've been very pleased with the quality of work.",
        "price": "$75",
        "introduction": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum",
        "backgroundChecked": true,
        "employees": 2,
        "paymentMethods": [
          "Cash", "Venmo"
        ],
        "faqs": [
          {
            "question": "What should the customer know about your pricing (e.g., discounts, fees)?",
            "answer": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
          },
          {
            "question": "What education and/or training do you have that relates to your work?",
            "answer": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
          }
        ],
        "reviews": [
          {
            "id": 1,
            "who": "Kristine M.",
            "stars": 5,
            "review":"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
            "date": "Apr 13, 2017",
            "reply": "thank you very much!! see you next weekend!!"
          },
          {
            "id": 2,
            "who": "Eric G.",
            "stars": 5,
            "review":"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
            "date": "Apr 19, 2017",
            "reply": "Thank you so much Eric! We loved cleaning your home! "
          }
        ]
      },
      {
        "title": "Perfect Shine Cleaner",
        "rating": 5,
        "reviews": 1,
        "yearsInBusiness": "1",
        "hires": 3,
        "latestReview": "Ana came is and did an amazing job cleaning before a family party. She was available on short notice and exceeded my expectations. Thanks Ana!",
        "price": "$90"
      },
      {
        "title": "Mendes cleaning services",
        "rating": 4.8,
        "reviews": 35,
        "yearsInBusiness": "5",
        "hires": 23,
        "latestReview": "Wonderful job! House was quite a mess and they did a great job! Scheduling them for regular cleanings! High recommend!!!",
        "price": "$90"
      }
    ],
    "questions": [
      {
        "id": "1",
        "question": "Number of bedrooms",
        "questionType": "MULTIPLE-CHOICE",
        "choices": [
          {"id": "1", "title": "1 bedroom"},
          {"id": "2", "title": "2 bedroom"},
          {"id": "3", "title": "3 bedroom"},
          {"id": "4", "title": "4 bedroom"},
          {"id": "5", "title": "5 bedroom"},
          {"id": "6", "title": "6 bedroom"}
        ]},
      {"id": "2", "question": "Number of bathrooms", "questionType": "MULTIPLE-CHOICE",
        "choices": [
          {"id": "1", "title": "1 bathroom"},
          {"id": "2", "title": "2 bathroom"},
          {"id": "3", "title": "3 bathroom"},
          {"id": "4", "title": "4 bathroom"},
          {"id": "5", "title": "5 bathroom"},
          {"id": "6", "title": "6 bathroom"}
        ]},
      {"id": "3", "question": "Cleaning type", "questionType": "MULTIPLE-CHOICE",
        "choices": [
          {"id": "1", "title": "Standard cleaning"},
          {"id": "2", "title": "Deep cleaning"},
          {"id": "3", "title": "Move out cleaning"}
        ]},
      {"id": "4", "question": "Extra services", "questionType": "MULTIPLE-ANSWER",
        "choices": [
          {"id": "1", "title": "Fridge cleaning"},
          {"id": "2", "title": "Oven cleaning"},
          {"id": "3", "title": "Window cleaning"},
          {"id": "4", "title": "Laundry"}
        ]},
      {"id": "5", "question": "Frequency", "questionType": "MULTIPLE-CHOICE",
        "choices": [
          {"id": "1", "title": "Just once"},
          {"id": "2", "title": "Every week"},
          {"id": "3", "title": "Every 2 weeks"},
          {"id": "4", "title": "Once a month"}
        ]}
    ]
  },
  {
    "id": "345", "title": "Pets",
    "serviceProviders": [],
    "questions": [],
    "services": [
    {"id": "1", "title": "Cat Grooming"},
    {"id": "2", "title": "Dog Training"},
    {"id": "3", "title": "Pet Boarding"},
    {"id": "4", "title": "Pet Sitting"},
    {"id": "5", "title": "Dog Walking"},
    {
      "id": "6",
      "title": "Dog Grooming",
      "serviceProviders": [
        {
          "title": "Lucky Dogs Daycare & Pet Spa",
          "rating": 4.4,
          "reviews": 21,
          "yearsInBusiness": "7",
          "hires": 17,
          "latestReview": "This weekend we  took her for a two-hour play time followed by ...",
          "price": "$35/Dog"
        },
        {
          "title": "Zenful Pawz LLC",
          "rating": 5,
          "reviews": 1,
          "yearsInBusiness": "1",
          "hires": 3,
          "latestReview": "! Sheâ€™s the best! I know my little Betty is in great care. We have been to many groomers & this is the best yet! We will continue to visit Izzy!",
          "price": "contact for price"
        },
        {
          "title": "Fluff & Buff Boutique",
          "rating": 4.8,
          "reviews": 35,
          "yearsInBusiness": "5",
          "hires": 23,
          "latestReview": "Kate really knows how to handle dogs. Her love of animals shines and she takes great pride in her work. My dog always looks great after her care.",
          "price": "contact for price"
        }
      ],
      "questions": [
        {"id": "1", "question": "Dog size",
          "questionType": "MULTIPLE-CHOICE",
          "choices": [
            {"id": "1", "title": "Extra small (1 - 15 lbs)"},
            {"id": "2", "title": "Small (16 - 25 lbs)"},
            {"id": "3", "title": "Medium (26 - 40 lbs)"},
            {"id": "4", "title": "Large (41 - 70 lbs)"},
            {"id": "5", "title": "Extra large (70+ lbs)"}
          ]},
        {"id": "2", "question": "Dog coat type",
          "questionType": "MULTIPLE-CHOICE",
          "choices": [
            {"id": "1", "title": "Smooth, skin-like coat"},
            {"id": "2", "title": "Wavy or curly coat"},
            {"id": "3", "title": "Silky coat with fine hair"},
            {"id": "4", "title": "Wiry, bristly coat"},
            {"id": "5", "title": "Thick, fluffy double-coat"}
          ]}
      ]},
    {"id": "7", "title": "Dog Daycare"},
    {"id": "8", "title": "Aquarium Services"},
    {"id": "9", "title": "Animal Training (Non-canine)"},
    {"id": "10", "title": "Behavior Modification (Non-canine)"}
  ]},
  {
    "id": "234", "title": "Wellness",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "456", "title": "Business",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "567", "title": "Events",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "678", "title": "Crafts",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "789", "title": "Design and Web",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "890", "title": "Legal",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "987", "title": "Personal",
    "serviceProviders": [],
    "questions": [],
    "services": []},
  {
    "id": "876", "title": "Photography",
    "serviceProviders": [],
    "questions": [],
    "services": []}
]