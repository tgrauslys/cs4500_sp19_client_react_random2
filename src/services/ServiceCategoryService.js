// Retrieves service category information
export default class ServiceCategoryService {
    // Holds instance of this class
    static instance = null;

    static getInstance() {
        // Allow only one instance
        if (ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }

    // Lambdas to make REST calls
    findServiceCategoryById = categoryId =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}api/categories/${categoryId}`)
            .then(response => response.json());
    findAllServiceCategories = () =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}api/categories`)
            .then(response => response.json());

    deleteServiceCategoryById = categoryId =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}api/categories/${categoryId}`, {
            method: 'DELETE',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            }
        });

    createServiceCategory = () =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}api/categories`, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                serviceCategoryName: "",
                services: []
                                 })}).then(response => response.json());

}