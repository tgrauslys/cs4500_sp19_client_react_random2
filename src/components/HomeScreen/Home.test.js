import React from 'react'
import ServiceCategoryService from '../../services/ServiceCategoryService'
import '../../services/ServiceCategoryService.mock'
import {BrowserRouter} from "react-router-dom";
import TestRenderer from 'react-test-renderer'
import ServiceCategories from "../../components/ServiceCategories";
import Home from "./Home";

const servCatServ = ServiceCategoryService.getInstance();

test('Render home screen correctly', () => {

    servCatServ.findAllServiceCategories().then(categories => {
        const testRenderer = TestRenderer.create(
            <BrowserRouter>
                <Home
                    pillServiceCategories={categories}
                />
            </BrowserRouter>
        );

        let tree = testRenderer.toJSON();

        expect(tree).toMatchSnapshot();


        const testInstance = testRenderer.root;


        // Test that all the service category pill buttons are there
        const delBtns = testInstance.findAllByProps({className: "nav-link btn-lg text-center"});
        expect(delBtns.length).toBe(5);
        expect(delBtns[0].props.children).toBe("Dog");

        // Test that the search fields are there
        const pageBtns = testInstance.findAllByProps({className: "input"});
        expect(pageBtns.length).toBe(2);


    });
});
