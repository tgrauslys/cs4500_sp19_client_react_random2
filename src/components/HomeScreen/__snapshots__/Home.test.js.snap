// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render home screen correctly 1`] = `
<div>
  <h2>
    Home
  </h2>
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <h1>
        Find professionals near you.
      </h1>
    </div>
    <div
      className="col-3 text-right"
    >
      <a
        href="#"
      >
        Sign up
      </a>
    </div>
    <div
      className="col-1"
    >
      <a
        href="#"
      >
        Log in
      </a>
    </div>
  </div>
  <br />
  <div>
    <ul
      className="nav nav-pills nav-fill wd-shadow wd-padding-20"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link btn-lg text-center"
          href="/categories/1722"
        >
          Dog
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link btn-lg text-center"
          href="/categories/1732"
        >
          Cave
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link btn-lg text-center"
          href="/categories/1742"
        >
          Pool
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link btn-lg text-center"
          href="/categories/2542"
        >
          Internet
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link btn-lg text-center"
          href="/categories/2652"
        >
          Beyond
        </a>
      </li>
    </ul>
  </div>
  <br />
  <div>
    <div>
      <div>
        <form
          className="form"
        >
          <div
            className="input-group mb-3"
          >
            <input
              className="input"
              name="username"
              onChange={[Function]}
              placeholder="Search for providers..."
              type="text"
            />
            <input
              className="input"
              name="zipcode"
              onChange={[Function]}
              placeholder="Zipcode"
              type="text"
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-success"
                onClick={[Function]}
                style={
                  Object {
                    "margin": "2px",
                  }
                }
                type="button"
              >
                Search
              </button>
            </div>
          </div>
        </form>
      </div>
      <div>
        <h3>
          Service Providers
        </h3>
        <table
          className="table"
        >
          <tbody />
        </table>
      </div>
    </div>
  </div>
  <br />
  <div>
    <div>
      <h2>
        ServiceNavigator
      </h2>
      <div>
        <div
          className="row"
        >
          <div
            className="col-3"
          >
            <table
              className="table"
            >
              <tbody>
                <td>
                  <a
                    href="/services-nav/categories/1722"
                    onClick={[Function]}
                  >
                    Dog
                  </a>
                </td>
                <td>
                  <a
                    href="/services-nav/categories/1732"
                    onClick={[Function]}
                  >
                    Cave
                  </a>
                </td>
                <td>
                  <a
                    href="/services-nav/categories/1742"
                    onClick={[Function]}
                  >
                    Pool
                  </a>
                </td>
                <td>
                  <a
                    href="/services-nav/categories/2542"
                    onClick={[Function]}
                  >
                    Internet
                  </a>
                </td>
                <td>
                  <a
                    href="/services-nav/categories/2652"
                    onClick={[Function]}
                  >
                    Beyond
                  </a>
                </td>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="col-9"
        />
      </div>
    </div>
  </div>
</div>
`;
